libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/SchemaDownloader.kt

```diff
@@ -44,26 +44,37 @@ object SchemaDownloader {
    * @param headers extra HTTP headers to send during introspection.
    */
   fun download(
-      endpoint: String? = null,
-      graph: String? = null,
-      key: String? = null,
-      graphVariant: String = "current",
+      endpoint: String?,
+      graph: String?,
+      key: String?,
+      graphVariant: String,
       registryUrl: String = "https://graphql.api.apollographql.com/api/graphql",
       schema: File,
       insecure: Boolean = false,
-      headers: Map<String, String> = emptyMap()
+      headers: Map<String, String> = emptyMap(),
   ) {
     var introspectionSchemaJson: String? = null
     var introspectionSchema: IntrospectionSchema? = null
     var gqlSchema: GQLDocument? = null
     when {
       endpoint != null -> {
-        introspectionSchemaJson = downloadIntrospection(
-            endpoint = endpoint,
-            headers = headers,
-            insecure = insecure,
-        )
-        introspectionSchema = introspectionSchemaJson.toIntrospectionSchema()
+        introspectionSchemaJson = try {
+          downloadIntrospection(
+              endpoint = endpoint,
+              headers = headers,
+              insecure = insecure,
+              includeDeprecatedInputFieldsAndArguments = true,
+          )
+        } catch (e: Exception) {
+          // Maybe the server doesn't support deprecated input fields / arguments, try without them
```

Comment:
Note for 4.x: we can do better than this and only retry if the server returned `response.errors`. This will avoid falling back to an "older" introspection if there is a transient network error.